plugins/module_utils/controller.py

```diff
@@ -81,8 +87,13 @@ def fail_wanted_one(self, response, endpoint, query_params):
         msg = f"Request to {display_endpoint} returned {response.json['count']} items, expected 1"
         raise EDAError(msg)
 
-    def get_exactly_one(self, endpoint, name=None, **kwargs):
-        return self.get_one_or_many(endpoint, name=name, allow_none=False, **kwargs)
+    def get_exactly_one(
+        self, endpoint, name=None, **kwargs
+    ) -> Optional[dict[str, bool]]:
+        result = self.get_one_or_many(endpoint, name=name, allow_none=False, **kwargs)
+        if isinstance(result, list):
```

Comment:
This case is handled here https://github.com/ansible/event-driven-ansible/pull/270/files#diff-4fd0f63c1f7e4adaee865c4b96a5b4bdae623dad664c81e1f5a777823342d829R137. So, if you're handling this corner case in this function I would suggest to clean up get_one_or_many() at this point.

I wouldn't raise a RuntimeError, but I would suggest using something similar self.fail_wanted_one(response, endpoint, new_kwargs.get("data")) or raise an EDAError with a more descriptive message for the user.

Reply:
I switched to fail_wanted_one.

Reply:
I you don't clean up the logic inside get_one_or_many() that handles the case when more than one resources are found, I do not think that your change will make any difference.